server/functions/framework/compiled_function.go

```diff
@@ -535,15 +537,23 @@ func (c *CompiledFunction) resolvePolymorphicReturnType(functionInterfaceTypes [
 	if !ok {
 		return returnType
 	}
-	// We can use the first polymorphic type that we find, since we can morph it into any type that we need.
+	// We can use the first polymorphic non-unknown type that we find, since we can morph it into any type that we need.
 	// We've verified that all polymorphic types are compatible in a previous step, so this is safe to do.
 	var firstPolymorphicType pgtypes.DoltgresType
 	for i, functionInterfaceType := range functionInterfaceTypes {
 		if _, ok = functionInterfaceType.(pgtypes.DoltgresPolymorphicType); ok {
 			firstPolymorphicType = originalTypes[i]
-			break
+			if _, isUnknownType := firstPolymorphicType.(pgtypes.UnknownType); !isUnknownType {
```

Comment:
Rather than make the check here, we should change
```
if _, ok = functionInterfaceType.(pgtypes.DoltgresPolymorphicType); ok
```
to
```
if _, ok = functionInterfaceType.(pgtypes.DoltgresPolymorphicType); ok && originalTypes[i].BaseID() != pgtypes.DoltgresTypeBaseID_Unknown
```
And remove this change here